lgc/patch/FragColorExport.cpp

```diff
@@ -358,36 +352,36 @@ Value *FragColorExport::run(Value *output, unsigned location, Instruction *inser
 //
 // @param outputTy : Type of fragment data output
 // @param location : Location of fragment data output
-ExportFormat FragColorExport::computeExportFormat(Type *outputTy, unsigned location) const {
-  GfxIpVersion gfxIp = m_pipelineState->getTargetInfo().getGfxIpVersion();
-  auto gpuWorkarounds = &m_pipelineState->getTargetInfo().getGpuWorkarounds();
+ExportFormat FragColorExport::computeExportFormat(Type *outputTy, const ColorExportFormat *colorExportFormat,
+                                                  unsigned location) const {
+  GfxIpVersion gfxIp = m_lgcContext->getTargetInfo().getGfxIpVersion();
+  auto gpuWorkarounds = &m_lgcContext->getTargetInfo().getGpuWorkarounds();
   unsigned outputMask = outputTy->isVectorTy() ? (1 << cast<VectorType>(outputTy)->getNumElements()) - 1 : 1;
-  const auto cbState = &m_pipelineState->getColorExportState();
-  const auto target = &m_pipelineState->getColorExportFormat(location);
+  const auto cbState = &m_colorExportState;
   // NOTE: Alpha-to-coverage only takes effect for outputs from color target 0.
   const bool enableAlphaToCoverage = (cbState->alphaToCoverageEnable && location == 0);
```

Comment:
Can you get rid of this dependence on `location`? That seems to be the only one that's left.